src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -494,10 +500,14 @@ function copy_to_coupled_boundary!(boundary_condition::BoundaryConditionCoupled{
         j_node = j_node_start
 
         for i in eachnode(solver)
-            for v in 1:size(u, 1)
-                boundary_condition.u_boundary[v, i, cell] = u[v, i_node, j_node,
-                                                              linear_indices[i_cell,
-                                                                             j_cell]]
+            for v in 1:size(boundary_condition.u_boundary, 1)
+                x = cache.elements.node_coordinates[:, i_node, j_node,
+                                                    linear_indices[i_cell, j_cell]]
+                converted_u = boundary_condition.coupling_converter(x,
+                                                                    u[:, i_node, j_node,
+                                                                      linear_indices[i_cell,
+                                                                                     j_cell]])
+                boundary_condition.u_boundary[v, i, cell] = converted_u[v]
             end
```

Comment:
If I am not mistaken, I think the inner loop is superfluous. What you are essentially doing here is to call the converter function on each node _once per variable_, and then picking off only one component of it.

Also, the `u[:, ...]` and `x[:, ...]` parts will allocate.

You also need to add something like
```julia
@unpack node_coordinates = cache.elements
@unpack coupling_converter, u_boundary = boundary_condition
```
somewhere above.


IMHO something like the following should work (please re-check, this might still contain bug).
```suggestion
            x = get_node_vars(node_coordinates, equations, solver, i_node, j_node, element_id)
            u_node = get_node_vars(u, equations, solver, i_node, j_node, element_id)
            converted_u = coupling_converter(x, u_node)
            boundary_condition.u_boundary[v, i, cell] = converted_u[v]
```

Reply:
Also, what I'm missing here is the ability to use the equation object from both the source and the target equations to compute the new target state. Do you know how we can add this?

Reply:
Just implemented your suggestions. It seems to work fine.